### DISCUSSION TITLE
useSWR is not a function

### DISCUSSION THREAD

**Discussion by yonayarin** [vercel/swr#2107] [Q&A]
# Bug report

## Description / Observed Behavior

Hi, lately I am facing issues with useSWR, it seems that something has been changed in the last versions and it is not a function anymore. 
<img width="548" alt="Screen Shot 2022-08-07 at 11 19 43" src="https://user-images.githubusercontent.com/98890583/183286456-5f3285f4-d29f-4e1d-9d95-c30fcc0f3ea2.png">

I got to it by using 
`import useSWRInfinite from 'swr/infinite';`

The problem starts with the last update from 0.5.6 -> 1.3.0

**Comment by huozhi** [comment]
Can you share entire codebase of the minimal reproduction? It feels like sth wrong with bundling that didn't resolve the default import properly

**Comment by yonayarin** [comment]
Unfortunately, I can share the entire codebase but I can share our usage...
```
 const fallbackData = useMemo(
    () =>
      loadDataFromSessionStorage(url, getNextKey as (data: unknown) => string),
    [url, getNextKey]
  );


  const { data, error, mutate, setSize } = useSWRInfinite<T>(
    (pageIndex, previousPageData) =>
      pageIndex === 0 ? url : getNextKey(previousPageData as T),
    fetcher,
    {
      fallbackData,
      onSuccess: (newData, key) => saveInStorage<T>(newData, key, getNextKey),
      revalidateOnMount: true,
      revalidateOnFocus: false,
      shouldRetryOnError: false,
      revalidateFirstPage: false,
    }
```
we are using common.js

**Comment by huozhi** [comment]
The issue is not related to the swr usage, more related to the bundle strategy. The module is loaded by `require` in commonjs but the default export is not been picked correctly.

Might be related to if you forcedly specify commonjs transform somewhere in your project.

**Reply by tzachbon** [reply]
@huozhi, the issue was resolved when following this comment:
https://github.com/vercel/swr/discussions/1867#discussioncomment-2219438
> Pinning swr to "1.1.2".

Any suggestion on why? The setup is a simple webpack and babel.

**Reply by huozhi** [reply]
Could you provide a minimal reproduction? simple webpack and babel can be any configuration..There might be some unexpected babel plugin configuration led to this issue

I'm not able to repro the issue in https://github.com/vercel/swr/discussions/1867#discussioncomment-2219438 with latest swr and next

**Reply by tzachbon** [reply]
@huozhi I created [this repo](https://github.com/tzachbon/swr-mf-error).
As you can see, the problem might be in Module Federation (SharedPlugin). Maybe you can give us more details regarding the changes.

EDIT:
I opened an issue to Webpack. 
https://github.com/webpack/webpack/issues/16125

It would be very helpful if you could share the difference between 1.1.2 to 1.2.0.
Thank you so much for your attention and participation.

**Reply by huozhi** [reply]
Thanks for providing the reproduction! The change brought into 1.2.0 is the esm assets in output. When the bundler is able to consume mjs (esm output) they will pick it up.

It seems not to relate to babel since it still doesn't work with babel loader removed. I wonder it's related to module federation plugin? And thanks for filing the webpack issue 🙏 I'll also follow the updates there

**Reply by tzachbon** [reply]
No problem, anytime.
Thanks again for your assistance!

**Reply by huozhi** [reply]
Changing from `'swr'` to `'swr/'` resolved the problem for me
https://github.com/webpack/webpack/issues/16125#issuecomment-1210447191

**Reply by yonayarin** [reply]
Thanks a lot for your help! for now excluding it from the MF dependencies seem to work also.
Hope that Webpack team will fix it soon 😄

**Comment by ZacButko** [comment]
Hi, would it be reasonable for SWR to provide a named export for infinite from swr root? We just stumbled into this issue, and it seems the deep modules cause issues where other packages do not.
